src/common/sourceUtils.ts

```diff
@@ -358,3 +358,52 @@ export function getSyntaxErrorIn(code: string): Error | void {
     return e;
   }
 }
+
+export function getBestSteppableExpressionAt(src: string, offset: number) {
+  const ast = parseProgram(src);
+  let other: Node | undefined;
+  let steppable: Node | undefined;
+  traverse(ast, {
+    enter: node => {
+      const asAcorn = node as AcornNode;
+      if (offset >= asAcorn.start && offset < asAcorn.end) {
+        if (node.type === 'CallExpression' || node.type === 'NewExpression') {
+          steppable = node;
+        } else {
+          other = node;
+        }
+      }
+    },
+  });
+
+  return steppable || other;
+}
+
+const notParensRe = /^[^()]+/g;
+
+/**
+ * Gets a "step into" target at the given offset in the line of code.
+ */
+export function getStepTargetInfo(line: string, offset: number) {
```

Comment:
Is this still going to return breakpoint locations that are not step into targets? I'm not sure